src/plugins/lti/LTI.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect, useContext } from "react"
+import shortid from "shortid"
+
+import UserContext from "~/Contexts/User"
+
+import Flex from "~/components/Flex"
+import Loader from "~/components/Loader"
+import { getTools, getPseudonym, sign, depseudonymizationFrame } from './utils'
+
+const LTI = ({ state, editable }) => {
+  const [tools, setTools] = useState({})
+  const [loading, setLoading] = useState(true)
+  const [idToken, setIdToken] = useState("")
+  const { store: user } = useContext(UserContext)
+
+  async function bootstrap() {
+    if (state.url.value) {
+      const iss = window.location.protocol + '//' + window.location.host + '/';
+      const pseudonym = await getPseudonym(user.id, state.templateId.value)
+      const current = new Date();
+      const name = encodeURI(depseudonymizationFrame(iss, pseudonym));
+      if(state.ltiVersion.value === '1.3.0') {
+        const request = {
+          iss,
+          name,
+          aud: state.clientId.value,
+          sub: pseudonym,
+          exp: current.getTime() + 3 * 60,
+          iat: current.getTime(),
+          nonce: shortid.generate() + shortid.generate(),
+          'https://purl.imsglobal.org/spec/lti/claim/message_type': state.ltiMessageType.value,
+          'https://purl.imsglobal.org/spec/lti/claim/roles': [
+            `http://purl.imsglobal.org/vocab/lis/v2/membership#${user.roles[0].name}`,
```

Comment:
It exist situations where the user can have more then one role. Is it important?